### PR TITLE
Repaired the Big batch mode for FFTW

### PR DESCRIPTION
**Pull request by Shanigen**
Big batch mode now works with FFTW. I also made some minor esthetic changes.

### CODE REVIEW

**Review by wentasah — CHANGES_REQUESTED** [review]
Can you change it so that formatting changes are not intermixed with functionality changes?

File: src/fft_fftw.cpp [hunk 1]
```diff
@@ -92,35 +92,35 @@ void Fftw::init(unsigned width, unsigned height, unsigned num_of_feats, unsigned
 #endif
 #ifdef BIG_BATCH
     // FFT inverse all scales
-    {
-        ComplexMat in_i_all(m_height, m_width, m_num_of_feats * m_num_of_scales);
-        cv::Mat out_i_all = cv::Mat::zeros(m_height, m_width, CV_32FC(m_num_of_feats * m_num_of_scales));
+    if (m_num_of_scales > 1) {
```
Comment (wentasah) [inline src/fft_fftw.cpp:95]: Why this if? For big batch, m_num_of_scales is always > 1.

File: src/fft_fftw.cpp [hunk 2]
```diff
@@ -179,8 +179,7 @@ void Fftw::inverse(ComplexMat &complex_input, MatScales &real_result)
     fftwf_complex *in = reinterpret_cast<fftwf_complex *>(complex_input.get_p_data());
     float *out = real_result.ptr<float>();
 
-    if (n_channels == 1)
-        fftwf_execute_dft_c2r(plan_i_1ch, in, out);
+    if (n_channels == 1) fftwf_execute_dft_c2r(plan_i_1ch, in, out);
```
Comment (wentasah) [inline src/fft_fftw.cpp:182]: Do not put this on a single line. I've changed the clang-forrmat configuration. Configure your tools to use the configuration from the repository.

**Review by wentasah — CHANGES_REQUESTED** [review]

File: src/fft_fftw.cpp (hunk 1 above)
Comment (Shanigen) [reply src/fft_fftw.cpp:95]: This is safety check, for the possibility that the someone would set the p_num_scales to 1 in kcf.h and wanted to do big batch mode. In that case it is unnecessary to create big batch plans.

Comment (wentasah) [reply src/fft_fftw.cpp:95]: So it is not safety check but unnecessary optimization. If the check is not there, the program will still work correctly, right?

Comment (Shanigen) [reply src/fft_fftw.cpp:95]: Yes. I worded that incorrectly.

**Comment by wentasah** [comment]
There is no need to close the pull request and open a new one. When you push a corrected branch e.g. `git push -f`, it will be visible here.

**Pull request closed without merging**